mllib/src/main/scala/org/apache/spark/mllib/feature/ChiSqSelector.scala

```diff
@@ -189,11 +232,21 @@ class ChiSqSelector @Since("1.3.0") (
    */
   @Since("1.3.0")
   def fit(data: RDD[LabeledPoint]): ChiSqSelectorModel = {
-    val indices = Statistics.chiSqTest(data)
-      .zipWithIndex.sortBy { case (res, _) => -res.statistic }
-      .take(numTopFeatures)
-      .map { case (_, indices) => indices }
-      .sorted
+    val chiSqTestResult = Statistics.chiSqTest(data)
+    val features = selectorType match {
+      case ChiSqSelectorType.KBest => chiSqTestResult
+        .zipWithIndex.sortBy { case (res, _) => -res.statistic }
```

Comment:
No just execute ` .zipWithIndex.sortBy { case (res, _) => -res.statistic }` no matter what before the match. I know it ends up being superfluous in one case but later we may want to retain sorted chi square results in the model. The other two cases need it anyway.